### ISSUE TITLE
Successful connection without peer's certificate when fail_if_no_peer_cert = true

### ISSUE THREAD

**Issue by kutashenko** [emqtt/emqttd#1423]
#### Environment

- OS: macOS
- Erlang/OTP: 
- EMQ: emqttd 2.3.2

#### Description

**Part of my configuration:**
listener.ssl.external.handshake_timeout = 15s
listener.ssl.external.keyfile = etc/certs/localhost.key
listener.ssl.external.certfile = etc/certs/localhost.crt
listener.ssl.external.cacertfile = etc/certs/ca.crt
listener.ssl.external.verify = verify_peer
listener.ssl.external.fail_if_no_peer_cert = true

**Connection by benchmark app:**
./emqtt_bench_pub -c 1 -i 10 -t bench/%i -s 256 -p 8883
20:25:22.350 [info] Application lager started on node nonode@nohost
20:25:22.351 [info] Application emqttc started on node nonode@nohost
20:25:22.351 [info] Application emqtt_benchmark started on node nonode@nohost
conneted: 1
sent(2001): total=1, rate=1(msg/sec)
sent(3002): total=2, rate=1(msg/sec)

**Comment by turtleDeng** [comment]
```
./emqtt_bench_pub -c 1 -i 10 -t bench/%i -s 256 -p 8883
09:48:01.725 [info] Application lager started on node nonode@nohost
09:48:01.727 [info] Application emqttc started on node nonode@nohost
09:48:01.727 [info] Application emqtt_benchmark started on node nonode@nohost
conneted: 1
sent(2001): total=1, rate=1(msg/sec)
sent(3001): total=2, rate=1(msg/sec)
sent(4001): total=3, rate=1(msg/sec)
sent(5001): total=4, rate=1(msg/sec)
sent(6001): total=5, rate=1(msg/sec)
sent(7001): total=6, rate=1(msg/sec)
sent(8001): total=7, rate=1(msg/sec)
sent(9001): total=8, rate=1(msg/sec)
sent(10001): total=9, rate=1(msg/sec)
sent(11001): total=10, rate=1(msg/sec)
sent(12001): total=11, rate=1(msg/sec)
sent(13001): total=12, rate=1(msg/sec)
sent(14001): total=13, rate=1(msg/sec)
sent(15001): total=14, rate=1(msg/sec)
client 1 EXIT: {shutdown,tcp_closed}
sent(16000): total=15, rate=1(msg/sec)
```

**Comment by HJianBo** [comment]
Hi, @kutashenko If you want to connect an SSL port, the `emqtt_bench_pub` should add the `-S` flag to enable it. like this:
```
/emqtt_bench_pub -c 1 -i 10 -t bench/%s -s 256 -S -p 8883
```
And then, use `--cerfile` & `--keyfile` flag to setup client's certificates

**Comment by HJianBo** [comment]
Hi, @kutashenko This issue has fixed in the v2.3.5.

The SSL upgrade will upgrade failed when some connection arrived.

EMQ Log:
```log
(emq@127.0.0.1)1> 10:42:37.063 [error] CRASH REPORT Process <0.1459.0> with 0 neighbours crashed with reason: {ssl_error,{options,{cacertfile,[]}}} in esockd_connection:upgrade/1 line 131
10:42:37.063 [error] Supervisor 'esockd_connection_sup - <0.1424.0>' had child connection started with emqttd_client:start_link([{max_publish_rate,0},{client_idle_timeout,30000},{client_enable_stats,false},{max_clientid_len,...},...]) at <0.1459.0> exit with reason {ssl_error,{options,{cacertfile,[]}}} in esockd_connection:upgrade/1 line 131 in context connection_crashed
```
Benchmark App log:
```log
$ ./emqtt_bench_pub -c 1 -i 10 -t bench/%i -s 256 -p 8883
10:42:37.029 [info] Application lager started on node nonode@nohost
10:42:37.030 [info] Application emqttc started on node nonode@nohost
10:42:37.030 [info] Application emqtt_benchmark started on node nonode@nohost
conneted: 1
client 1 EXIT: {shutdown,tcp_closed}
```
